packages/regenesis-surgery/src/download-solc.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Optimism PBC
+ */
+
+import axios from 'axios'
+import { access } from 'fs/promises'
+import path from 'path'
+import * as fs from 'fs'
+
+import {
+  LOCAL_SOLC_DIR,
+  EMSCRIPTEN_BUILD_LIST,
+  EMSCRIPTEN_BUILD_PATH,
+} from './constants'
+
+export const downloadSolc = async (version: string) => {
+  console.log(`Downloading ${version}`)
+  const res = await axios.get(EMSCRIPTEN_BUILD_LIST)
+  const list = await res.data.builds
+
+  let target
+  for (const entry of list) {
+    const longVersion = `v${entry.longVersion}`
+    if (version === longVersion) {
+      target = entry
+    }
+  }
+  if (!target) {
+    throw new Error(`Cannot find compiler version ${version}`)
+  }
+
+  const file = path.join(LOCAL_SOLC_DIR, target.path)
+
+  try {
+    await access(file, fs.constants.F_OK)
+    console.log(`${target.path} already downloaded`)
+  } catch (e) {
+    console.log(`Downloading ${target.path}`)
+    const bin = await axios.get(`${EMSCRIPTEN_BUILD_PATH}/${target.path}`)
```

Comment:
i manually downloaded these from https://binaries.soliditylang.org/bin/soljson-v0.6.12+commit.27d51765.js and the like

this path looks like it's https://binaries.soliditylang.org/emscripten-wasm32/soljson-v0.6.12+commit.27d51765.js which isn't a hosted file, am i missing a string operation somewhere?

Reply:
There is no `soljson` prefix in the path, you can try running it and it does work

Reply:
just ran it and it does! i missed that the list.json is also under the emscripten-wasm32 namespace so the final link is actually https://binaries.soliditylang.org/emscripten-wasm32/solc-emscripten-wasm32-v0.7.6+commit.7338295f.js

@tynes what's the rationale for using these versions instead of /bin path? for my own knowledge! 

also this is mergeable after adding the one line in the compiler constant

Reply:
I believe that the emscripten builds are faster, I didn't do a comparison but that is my understanding from reading the docs